common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDB.java

```diff
@@ -197,6 +201,23 @@ public void delete(Class<?> type, Object naturalKey) throws Exception {
     };
   }
 
+  @Override
+  public <T> KVStoreView<T> viewWithCondition(Class<T> type, Predicate<T> condition) throws Exception {
+    return new KVStoreView<T>() {
+      @Override
+      public Iterator<T> iterator() {
+        try {
+          Iterator<T> base = new LevelDBIterator<>(type, LevelDB.this, this);
+          Iterable<T> iterable = () -> base;
```

Comment:
Hm, there's no easier way to filter an Iterator by Predicate? maybe not. But, nit, I think you need to return a new iterator each time in the lambda here, not return `base` every time, technically.

Reply:
Yes. I’ve tried to modify the `hasNext` of `LevelBbIterator`, but the predicate cannot be handled in it since it uses byte[] only (not deserialized to T).

Reply:
> you need to return a new iterator each time

Only `LevelDBIterator` could be accepted here since in `KVStoreView.closeableIterator()`, it casts the iterator to `KVStoreIterator`:
```java
  public KVStoreIterator<T> closeableIterator() throws Exception {
    return (KVStoreIterator<T>) iterator();
  }
```
So I use a stream API instead of new a iterator. Do I misunderstand here?

Reply:
https://stackoverflow.com/questions/24511052/how-to-convert-an-iterator-to-a-stream

Reply:
What's the disadvantage of above approach? I suddenly recognize that I could new a iterator which implements `KVStoreIterator` to wrap that `base`. I will try.

Reply:
Updated